core/types/crosslink.go

```diff
@@ -4,51 +4,62 @@ import (
 	"math/big"
 	"sort"
 
-	"github.com/ethereum/go-ethereum/rlp"
-
 	"github.com/ethereum/go-ethereum/common"
-
+	"github.com/ethereum/go-ethereum/rlp"
 	"github.com/harmony-one/harmony/block"
 )
 
 // CrossLink is only used on beacon chain to store the hash links from other shards
+// signature and bitmap correspond to |blockNumber|parentHash| byte array
+// Captial to enable rlp encoding
 type CrossLink struct {
-	ChainHeader *block.Header
+	ParentHashF  common.Hash
+	BlockNumberF *big.Int
+	SignatureF   [96]byte //aggregated signature
+	BitmapF      []byte   //corresponding bitmap mask for agg signature
+	ShardIDF     uint32   //need first verify signature on |blockNumber|blockHash| is correct
+	EpochF       *big.Int //need first verify signature on |blockNumber|blockHash| is correct
```

Comment:
This should be the parentEpoch as you used it to retrieve committee.

Reply:
fixed